src/Library.qs

```diff
@@ -88,11 +89,136 @@
         targetRegister : Qubit[]
     )
     : Unit is Adj + Ctl {
-        (ControlledOnInt(address, ApplyPauliFromBitString(PauliX, true, value, _)))
+        (ControlledOnInt(address, ApplyPauliFromBitString(PauliX, true, value,_)))
             (addressRegister!, targetRegister);
     }
 
 ///////////////////////////////////////////////////////////////////////////
 // UTILITY FUNCTIONS
 ///////////////////////////////////////////////////////////////////////////
+
+
+
+//BUCKET-BRIGADE
+///////////////////////////////////////////////////////////////////////////
+// PUBLIC API
+///////////////////////////////////////////////////////////////////////////
+    
+     
+   /// # Summary
+   /// Type representing a generic BBQRAM type.
+   /// # Input
+   /// ## Lookup
+   /// The named operation that will look up data from the BBQRAM. 
+   /// ## AddressSize
+   /// The size (number of bits) needed to represent an address for the QRAM.
+   /// ## DataSize
+   /// The size (number of bits) needed to represent a data value for the QRAM.
+  newtype BBQRAM = (
+   AddressSize : Int, 
+   DataSize : Int
+   );
+
+  /// # Summary
+  /// Creates an instance of an bucket-brigade QRAM given the data it needs to store.
+  /// # Input
+  /// ## dataValues
+  /// An array of tuples of the form (address, value) where the address is the address 
+  /// of the auxillary register to which the memory register is connected via CNOT gate and 
+  /// value is the data stored (either 0 or 1) in the memory register. 
+  /// # Output
+  /// BBRQRAM type
+  function BBQRAMOracle(dataValues : ((Int, Bool)[])) : BBQRAM{   
+    let largestAddress = Microsoft.Quantum.Math.Max(
+    Microsoft.Quantum.Arrays.Mapped(Fst<Int, Bool>, dataValues));
+        let addressx = Mapped(Fst<Int, Bool>, dataValues);
+        
+        //let bbqrams =Mapped(BB, addressx);
+
+        return Default<BBQRAM>()
+           // w/ LookupBB <- bbqrams
```

Comment:
You need this line, as well as a readout default value assignment for the writing operation for the QRAM type.